tools/client_package_initializer.py

```diff
@@ -0,0 +1,157 @@
+import argparse
+import os
+import subprocess
+from typing import Dict, Tuple
+
+
+class Setup:
+    NAME_KEY = "name"
+    DESCRIPTION_KEY = "description"
+    SETUP_REQUIRES_KEY = "setup_requires"
+    VCVERSIONER_KEY = "vcversioner"
+    AUTHOR_KEY = "author"
+    AUTHOR_EMAIL_KEY = "author_email"
+    URL_KEY = "url"
+    KEYWORDS_KEY = "keywords"
+    INSTALL_REQUIRES_KEY = "install_requires"
+    PACKAGES_KEY = "packages"
+    INCLUDE_PACKAGE_DATA_KEY = "include_package_data"
+    LONG_DESCRIPTION_KEY = "long_description"
+    PYTHON_REQUIRES_KEY = "python_requires"
+    all = [NAME_KEY, DESCRIPTION_KEY, SETUP_REQUIRES_KEY, VCVERSIONER_KEY, AUTHOR_KEY, AUTHOR_EMAIL_KEY, URL_KEY,
+           KEYWORDS_KEY, INSTALL_REQUIRES_KEY, PACKAGES_KEY, INCLUDE_PACKAGE_DATA_KEY, LONG_DESCRIPTION_KEY,
+           PYTHON_REQUIRES_KEY]
+
+    FORMAT = """
+import setuptools
+
+setuptools.setup(
+    name="{name}",
+    description="{description}",
+    setup_requires={setup_requires},
+    vcversioner={vcversioner},
+    author="{author}",
+    author_email="{author_email}",
+    url="{url}",
+    keywords={keywords},
+    install_requires={install_requires},
+    packages={packages},
+    include_package_data={include_package_data},    
+    python_requires='>=3.6',
+    long_description=open('README.md').read(),
+    long_description_content_type='text/markdown',
+    classifiers=[
+        'Development Status :: 3 - Alpha',
+        'Intended Audience :: Developers',
+        'Intended Audience :: Information Technology',
+        'License :: OSI Approved :: Apache Software License',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.6',
+        'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
+        'Programming Language :: Python :: 3.9',
+    ]
+)
+"""
+
+
+class SetupInitializer:
+    DEFAULT_SETUP_PATH = "setup.py"
+    DEFAULT_VERSION_PATH = "version.txt"
+    DEFAULT_README_PATH = "README.md"
+
+    def __init__(self, project_path: str, url: str) -> None:
+        self._project_path = project_path
+        self._setup_path = os.path.join(project_path, self.DEFAULT_SETUP_PATH)
+        self._url = url
+        self._version = None
+        self._setup_data = {"setup_requires": ['vcversioner'],
+                            "vcversioner": {'vcs_args': ['git', 'describe', '--tags', '--long']}}
+        self._tests_require = ""
+
+        with open("MANIFEST.in", "w") as f:
+            f.write("include README.md")
+
+    def _load_setup(self) -> None:
+        """
+        Load setup.py arguments as python dict
+        """
+        data = ""
+
+        with open(self._setup_path) as f:
+            lines = f.readlines()
+
+        for line in lines:
+            if "setup(" in line:
+                data += "SETUP = dict(\n"
+            else:
+                data += line
```

Comment:
I don't understand what's going on over here.. seems like an hack :(
Isn't that possible that that setup.py includes other not-dict related lines?
In addition, when do you stop loading lines? what happens if `setup(` doesn't exist? Is that naive loading just works?

Reply:
@YuviGold  The signature of setup is `def setup(**attrs)` so by definition it's acting like python dict.
My assumption was that if the  `setup( ` keyword is missing there was a problem generating client code and we didn't need to proceed the building process .
There is one library that does the same (parsing setup.py) but it also uses hack and assumptions on setup.py file.
We can discuss this issue offline if you will like.